### PR TITLE
REFACTOR ALL THE YHINGZ!

### PR DESCRIPTION
**Pull request by tiwiz**
Better? :-)

### CODE REVIEW

File: android/src/main/java/com/google/samples/apps/iosched/droidconitaly15/JsonTransformer.java [hunk 1]
```diff
@@ -46,6 +46,7 @@
     private static final Set<String> EXCLUDE_SESSIONS;
     private static final HashMap<String, String> ROOMS_TO_COLORS;
     private static final String DEFAULT_ROOM_COLOR;
+    private static final int NUMBER_OF_ROOMS = 8;
```
Comment (Takhion) [inline android/src/main/java/com/google/samples/apps/iosched/droidconitaly15/JsonTransformer.java]: Terrible, terrible idea! Keep it close to where it's initialized, trust me.

File: android/src/main/java/com/google/samples/apps/iosched/droidconitaly15/JsonTransformer.java [hunk 2]
```diff
@@ -266,10 +267,14 @@ private static void transformJson(
      * MOAR COLORZ PLS!
      */
     private static String getColorFrom(String roomName) {
-        String roomColor = DEFAULT_ROOM_COLOR;
+        if (roomName == null) {
+            return DEFAULT_ROOM_COLOR;
+        }
+
+        String roomColor = ROOMS_TO_COLORS.get(roomName.toLowerCase());
```
Comment (Takhion) [inline android/src/main/java/com/google/samples/apps/iosched/droidconitaly15/JsonTransformer.java]: While we're here you might also want to do `roomName.trim().toLowerCase()`, but again, nitpicking...

File: android/src/main/java/com/google/samples/apps/iosched/droidconitaly15/JsonTransformer.java (hunk 1 above)
Comment (rock3r) [inline android/src/main/java/com/google/samples/apps/iosched/droidconitaly15/JsonTransformer.java]: Yeah, wouldn't extract it as a constant either.

File: android/src/main/java/com/google/samples/apps/iosched/droidconitaly15/JsonTransformer.java (hunk 2 above)
Comment (rock3r) [inline android/src/main/java/com/google/samples/apps/iosched/droidconitaly15/JsonTransformer.java]: Yes. That. Also remember to change the `HashMap` initialisation with lowercase room names!

**Comment by rock3r** [comment]
selfie-0

`toLowerCase()` and what @Takhion said about the `NUMBER_OF_ROOMS`, then can be merged.

**Comment by rock3r** [comment]
Hmm ok the lowercase stuff is already done.

**Comment by Takhion** [comment]
Should add a Locale to `toLowerCase()` as it might yield different results but MAYBE ANOTHER DAY. #yolo

**Pull request merged**